lib/screens/cash_home/cash_home.dart

```diff
@@ -14,6 +14,9 @@ class CashHomeScreen extends StatefulWidget {
 
 
 void onChange(CashWalletViewModel viewModel) {
+  if (!viewModel.isListeningToBranch) {
+    viewModel.listenToBranch();
```

Comment:
I think you have a race condition here. If user already has a wallet, a switch community to a default community will be immediately. So he'll join the default community